pages/docs/reference/whatsnew14.md

```diff
@@ -574,3 +574,363 @@ Complete the initial configuration, and when you add a new dependency to `cocoap
+        val result = numbers.map { it * 2 }.shuffled().take(5)
+        println(result.toList()) //five random even numbers below 100
+    //sampleEnd
+    }
+    ```
+    </div>
+
+* `onEachIndexed()` counterpart have been added for [onEach()](/api/latest/jvm/stdlib/kotlin.collections/on-each.html).
+The operation that it applies to the collection elements has the element index as a parameter.
+
+    <div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+    
+    ```kotlin
+    fun main() {
+    //sampleStart
+        val list = mutableListOf("a", "b", "c", "d").onEachIndexed {
+            index, item -> println(index.toString() + ":" + item)
+        }
+    //sampleEnd
+    }
+    ```
+    </div>
+
+* New `*OrNull()` counterparts `randomOrNull()`, `reduceOrNull()`, and `reduceIndexedOrNull()`. 
+empty collections.
+    <div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+    
+    ```kotlin
+    fun main() {
+    //sampleStart
+         val empty = emptyList<Int>()
+         empty.reduceOrNull { a, b -> a + b }
+         //empty.reduce { a, b -> a + b } // Exception: Empty collection can't be reduced.
+    //sampleEnd
+    }
+    ```
+    </div>
+
+* `runningFold()` and `runningReduce()`, similarly to`fold()` and `reduce()`, apply the given operation to the collection
+elements subsequently; the difference is that they return the whole sequence of intermediate results.
+
+    <div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+    
+    ```kotlin
+    fun main() {
+    //sampleStart
+        val numbers = mutableListOf(0, 1, 2, 3, 4, 5)
+        val runningReduceSum = numbers.runningReduce() { sum, item -> sum + item} // [0, 1, 3, 6, 10, 15]
+        val runningFoldSum = numbers.runningFold(10) { sum, item -> sum + item} // [10, 10, 11, 13, 16, 20, 25]
+    //sampleEnd
+    }
+    ```
+    </div>
+
+* `sumOf()` takes a selector function and returns a sum of its values on all elements of a collection.
+`sumOf()` can produce sums of the types `Int`, `Long`, `Double`, `UInt`, `ULong`. On the JVM, `BigInteger` and `BigDecimal` are also available.
+
+    <div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+    
+    ```kotlin
+    data class OrderItem(val name: String, val price: Double, val count: Int)
+    
+    fun main() {
+    //sampleStart
+        val order = listOf<OrderItem>(
+            OrderItem("Cake", price = 10.0, count = 1),
+            OrderItem("Coffee", price = 2.5, count = 3),
+            OrderItem("Tea", price = 1.5, count = 2))
+    
+        val total = order.sumOf { it.price * it.count} // Double
+        val count = order.sumOf { it.count } // Int
+    //sampleEnd
+        println("You've ordered $count items that cost $total in total")
+    }
+    ```
+    </div>
+
+* `min()` and `max()` functions have been renamed to `minOrNull()` and `maxOrNull()` to comply with the naming
```

Comment:
minBy and maxBy were also renamed to minByOrNull and maxByOrNull correspondingly.
minWith and maxWith were also renamed to minWithOrNull and maxWithOrNull correspondingly.